airflow/www/security.py

```diff
@@ -44,6 +46,24 @@
 }
 
 
+class DefaultSessionInterface(SecureCookieSessionInterface):
+    """
+    Default cookie session interface.
+    This prevents creating flask sessions on REST API requests.
+    """
+
+    def save_session(self, *args, **kwargs):
+        """Prevent creating session from REST API requests."""
+        if g.get('login_from_api'):
```

Comment:
Maybe we should do check here `if request.url.startswith ("/ api")`? This will then simplify the code a bit. And it will also prevent discovery endpoints(non-login-required) from generating sessions.

Reply:
> Maybe we should do check here `if request.url.startswith ("/api/")`? This will then simplify the code a bit. And it will also prevent discovery endpoints(non-login-required) e.g. `/healtth`, `/version` from creating sessions.

And I think using this will also get rid of 'login_from_api'. Thanks! testing